torch/_inductor/fx_passes/mkldnn_fusion.py

```diff
@@ -655,6 +655,15 @@ def _recover_linear():
         def reshape_linear_reshape_pattern(match, *args, **kwargs):
             reshape_1 = kwargs.get("reshape_1")
             reshape_2 = kwargs.get("reshape_2")
+            assert len(reshape_1) == 2
+            dynamic_shapes = not all(
+                isinstance(x, int) for x in ([reshape_1[0]] + reshape_2[:-1])
+            )
+            if dynamic_shapes:
+                # For dynamic shapes, we are unsafe to compair the shapes for reshape_1 and reshape_2
+                # since they can be changed during runtime
+                return
+
             graph = match.graph
             node = match.output_node()
             if reshape_1[0] == reduce(lambda x, y: x * y, reshape_2[:-1]):
```

Comment:
How does `mkldnn._linear_pointwise.default` produce the output of the same shape as the one from `reshape_2`?

Reply:
Also checks
`linear's input's shape[:-1] == reshape_2[:-1]` and `product(reshape_2[:-1]) == reshape_1[0]`